ethereum/contracts/LightClientBridge.sol

```diff
@@ -73,12 +77,34 @@ contract LightClientBridge {
      * @dev If the validatorSetRegistry should be initialised with 0 entries, then input
      * 0x00 as validatorSetRoot
      * @param _validatorRegistry The contract to be used as the validator registry
+     * @param _mmrVerification The contract to be used for MMR verification
      */
-    constructor(ValidatorRegistry _validatorRegistry) {
+    constructor(ValidatorRegistry _validatorRegistry, MMRVerification _mmrVerification) {
         validatorRegistry = _validatorRegistry;
+        mmrVerification = _mmrVerification;
         currentId = 0;
     }
 
+    /**
+     * @notice Executed by the incoming channel in order to verify commitment
+     * @param beefyMerkleLeaf contains the merkle leaf to be verified
+     * @param beefyMerkleProof contains the merkle proof to verify against
+     */
+    function verifyBeefyMerkleLeaf(
+        bytes32 beefyMerkleLeaf,
```

Comment:
Thinking a bit more about it, I guess the lightClientBridge should not be concerned with the structure of the leaf at this stage.

It needs the full leaf to apply the validator set changes when applying the commitments, but here it would be more efficient and simpler to indeed just pass the hash of the leaf. That reduces call data, keccak hashing is super cheap on Ethereum.

Reply:
So `bytes32` would be correct.

Reply:
Rococo is using keccak, as discussed on Slack. For reference: https://github.com/paritytech/polkadot/blob/master/runtime/rococo/src/lib.rs#L781

Reply:
We do too: https://github.com/Snowfork/polkadot/blob/4050e97d2ddee159eddc354d8bec04d284f83383/runtime/rococo/src/lib.rs#L678

Reply:
as above